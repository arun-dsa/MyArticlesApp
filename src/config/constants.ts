export const NAVIGATION_LINKS: { [key: string]: string } = {
  ROOT: "/",
  ARTICLES: "/articles",
  ARTICLE_DETAILS: "/articles/:id",
};
