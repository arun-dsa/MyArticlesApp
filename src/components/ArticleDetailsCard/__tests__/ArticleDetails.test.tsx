import { ReactNode } from "react";
import { screen, render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { routeWrapper } from "@utils/test_utils";
import { ARTICLES } from "@tests/mockData";
import ArticlesDetailsCard from "..";
import { NAVIGATION_LINKS } from "@config/constants";

describe("Article details card test", () => {
  const article = ARTICLES[0];
  test("Test render article details", () => {
    const ArticleRoute: ReactNode = routeWrapper([
      {
        path: NAVIGATION_LINKS.ROOT,
        element: <ArticlesDetailsCard article={article} />,
      },
    ]);

    render(ArticleRoute);

    expect(
      screen.getByTestId(`article-details-title-${article.id}`)
    ).toBeInTheDocument();
    expect(
      screen.getByTestId(`article-details-title-${article.id}`)
    ).toHaveTextContent(article.title);

    expect(
      screen.getByTestId(`article-details-summary-${article.id}`)
    ).toBeInTheDocument();
    expect(
      screen.getByTestId(`article-details-summary-${article.id}`)
    ).toHaveTextContent(article.summary);

    expect(
      screen.getByTestId(`article-details-full-text-${article.id}`)
    ).toBeInTheDocument();
    expect(
      screen.getByTestId(`article-details-full-text-${article.id}`)
    ).toHaveTextContent(article.fullText);

    expect(
      screen.getByTestId(`share-article-${article.id}`)
    ).toBeInTheDocument();
  });

  test("Test snapshot of article details", () => {
    const ArticleRoute: ReactNode = routeWrapper([
      { path: "/", element: <ArticlesDetailsCard article={article} /> },
    ]);

    const container = render(ArticleRoute);
    expect(container).toMatchSnapshot();
  });
});
