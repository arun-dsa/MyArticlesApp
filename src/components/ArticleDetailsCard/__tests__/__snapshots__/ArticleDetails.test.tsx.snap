// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article details card test > Test snapshot of article details 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-16c4hvi-MuiStack-root"
        data-testid="article-details-card-1"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-18r8f8u-MuiTypography-root"
            data-testid="article-details-title-1"
          >
            Exploring the Wonders of the Great Barrier Reef
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              cursor="pointer"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </h5>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-5s066j-MuiTypography-root"
            data-testid="article-details-summary-1"
          >
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1rsw7ys-MuiTypography-root"
            data-testid="article-details-full-text-1"
          >
            The Great Barrier Reef, stretching over 2,300 kilometers along the coast of Queensland, Australia, is a marine wonderland. Home to thousands of species of fish, coral, and marine life, it is the largest coral reef system in the world. Snorkelers and divers flock here to witness its vibrant colors and diverse ecosystems. However, it faces threats from climate change and pollution, making conservation efforts crucial to its preservation.
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1yuhvjn"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-13pww1o-MuiButtonBase-root-MuiButton-root"
            data-testid="share-article-1"
            href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
            tabindex="0"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
                />
              </svg>
            </span>
            Share
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-16c4hvi-MuiStack-root"
      data-testid="article-details-card-1"
    >
      <div
        class="MuiStack-root css-nen11g-MuiStack-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-18r8f8u-MuiTypography-root"
          data-testid="article-details-title-1"
        >
          Exploring the Wonders of the Great Barrier Reef
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            cursor="pointer"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </h5>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-5s066j-MuiTypography-root"
          data-testid="article-details-summary-1"
        >
          Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1rsw7ys-MuiTypography-root"
          data-testid="article-details-full-text-1"
        >
          The Great Barrier Reef, stretching over 2,300 kilometers along the coast of Queensland, Australia, is a marine wonderland. Home to thousands of species of fish, coral, and marine life, it is the largest coral reef system in the world. Snorkelers and divers flock here to witness its vibrant colors and diverse ecosystems. However, it faces threats from climate change and pollution, making conservation efforts crucial to its preservation.
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-13pww1o-MuiButtonBase-root-MuiButton-root"
          data-testid="share-article-1"
          href="mailto:change-it-to-email-to-share@gmail.com?subject=Read this article - Exploring the Wonders of the Great Barrier Reef&body=Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.. Read the full article here http://localhost:3000/articles/1"
          tabindex="0"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShareIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92"
              />
            </svg>
          </span>
          Share
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
